elephant/cell_assembly_detection.py

```diff
@@ -0,0 +1,1182 @@
+        if elem in selection:
+            # find the element that was already in the list
+            pre = selection.index(elem)
+
+            if pre_pruning_assembly[i]['pvalue'][-1] <= \
+                    assembly[pre]['pvalue'][-1]:
+                # if the new element has a p-value that is smaller
+                # than the one had previously
+                selection[pre] = elem
+                # substitute the prev element in the selection with the new
+                assembly[pre] = pre_pruning_assembly[i]
+                # substitute also in the list of the new assemblies
+        if elem not in selection:
+            selection.append(elem)
+            assembly.append(pre_pruning_assembly[i])
+
+    # number of assemblies filtered out is equal to the difference
+    # between the pre and post pruning size
+    n_filtered_assemblies = nns - len(assembly)
+
+    return assembly, n_filtered_assemblies
+
+
+def _subgroup_pruning_step(pre_pruning_assembly):
+    """
+    Removes assemblies which are already all included in a bigger assembly
+
+    Parameters
+    ----------
+    pre_pruning_assembly : list
+        contains the assemblies filtered by the significance value
+
+    Returns
+    --------
+    final_assembly : list
+        contains the assemblies filtered by inclusion
+
+    """
+
+    # reversing the semifinal_assembly makes the computation quicker
+    # since the assembly are formed by agglomeration
+    pre_pruning_assembly_r = list(reversed(pre_pruning_assembly))
+    nns = len(pre_pruning_assembly_r)
+    # boolean list with the selected assemblies
+    selection = [True for _ in range(nns)]
+
+    for i in range(nns):
+        # check only in the range of the already selected assemblies
+        if selection[i]:
+            a = pre_pruning_assembly_r[i]['neurons']
+            for j in range(i + 1, nns):
+                if selection[j]:
+                    b = pre_pruning_assembly_r[j]['neurons']
+                    # check if a is included in b or vice versa
+                    if set(a).issuperset(set(b)):
+                        selection[j] = False
+                    if set(b).issuperset(set(a)):
+                        selection[i] = False
+                    # only for the case in which significance_pruning=False
+                    if set(a) == set(b):
+                        selection[i] = True
+                        selection[j] = True
+
+    assembly_r = []
+
+    # put into final_assembly only the selected ones
+    for i in range(nns):
+        if selection[i]:
+            assembly_r.append(pre_pruning_assembly_r[i])
+
+    assembly = list(reversed(assembly_r))
+
+    return assembly
+
+
+def _raise_errors(data, maxlag, alpha, min_occ, size_chunks, max_spikes):
+    """
+    Returns errors if the parameters given in input are not correct.
+
+    Parameters
+    ----------
+    data : numpy array
+        list of already binned spiketrains in listed form
```

Comment:
This is the input you give in the beginning right? So the type is then a `BinnedSpikeTrain` object, not numpy array or list.